testsuite/tests/apicast/policy/on_failed/test_onfailed.py

```diff
@@ -53,7 +53,7 @@ def test_on_failed_policy(application, status_code):
     Sends request to apicast and check that the returned code
     is the expected one as per `status_code`
     """
-    api_client = application.api_client()
+    api_client = application.api_client(disable_retry_status_list=(503,))
```

Comment:
hmm, and how to distinguish between setup delay and desired behavior. This seems poisonous

Reply:
That's the problem of the entire test as we are checking only 503 but not its cause. @palmieric Could we set up the maintenance mode to return a different status code so the initial setup 503 is distinguishable?

Reply:
Agree with Marian, you can't distinguish between 503 returned by chance and by actual behavior.

Reply:
@pehala the tests in place are both for:
- default configuration (apicast returns 503 if a policy fails/is skipped)
- custom configuration (apicast returns 500)
We can't bypass the default behavior or a possible configuration is lacking.

Reply:
in that case, we cannot be sure if the test passes with the default behaviour regardless of the ignore as it will retry and then still return 503.

How about we first set up service without policies, wait for it to become ready and then add the policy, in that case the route and everything should already be in place and we know that the 503 is from the policy.

Another solution would be checking the content of the response if it contains stuff related to the policy, like some header or seomething.

Reply:
it is possible to distinguish whether response came from apicast (openresty) or not. Custom backoff based retry on error code **and** "server" header seems to be a choice

Reply:
The question is how to modify it and who should do that (and if reuse this PR) I'll keep the decision on you two :-)

Reply:
@mganisin I already changed the PR to backoff on non-apicast answer, could you please re-review it?